config/tools/config_generator.cpp

```diff
@@ -0,0 +1,897 @@
+                set_param_to_config_file(raft_endpoint_key, raft_endpoint_val);
+                std::string log_level_key = atomizer_name + "_loglevel";
+                std::string log_level_val = _default_debug_level;
+                set_log_level(tmpl_atomizer_log_level, log_level_val);
+                set_param_to_config_file(log_level_key, log_level_val);
+            }
+        } else if(type == watchtower_count_key) {
+            // Create Watchtowers
+            for(size_t i = 0; i < number; i++) {
+                std::string watchtower_name = "watchtower" + std::to_string(i);
+                std::string client_endpoint_key
+                    = watchtower_name + "_client_endpoint";
+                std::string client_endpoint_val
+                    = cbdc::network::localhost + ":"
+                    + std::to_string(get_open_port());
+                set_param_to_config_file(client_endpoint_key,
+                                         client_endpoint_val);
+                std::string internal_endpoint_key
+                    = watchtower_name + "_internal_endpoint";
+                std::string internal_endpoint_val
+                    = cbdc::network::localhost + ":"
+                    + std::to_string(get_open_port());
+                set_param_to_config_file(internal_endpoint_key,
+                                         internal_endpoint_val);
+                std::string log_level_key = watchtower_name + "_loglevel";
+                std::string log_level_val = _default_debug_level;
+                set_log_level(tmpl_watchtower_log_level, log_level_val);
+                set_param_to_config_file(log_level_key, log_level_val);
+            }
+        } else {
+            std::cout << "Warning: Unrecognized component type, " << type
+                      << ", in Atomizer configuration generation."
+                      << std::endl;
+        }
+    }
+
+    void config_generator::load_template(
+        const std::string filename,
+        std::map<std::string, std::string>& config_map) {
+        std::ifstream file(filename);
+        assert(file.good());
+        std::string line;
+        while(std::getline(file, line)) {
+            std::istringstream line_stream(line);
+            std::string key;
+            if(std::getline(line_stream, key, '=')) {
+                std::string value;
+                if(std::getline(line_stream, value)) {
+                    config_map.emplace(key, value);
+                }
+            }
+        }
+    }
+
+    void config_generator::write_generated_config_to_file(
+        const std::string& _config_file) {
+        std::ofstream outFile;
+        outFile.open(_config_file);
+        outFile << m_new_config.str();
+        outFile.close();
+    }
+
+    // This method assumes project root is "opencbdc-tx" and build dir is
+    // "build"
+    [[nodiscard]] auto
+    config_generator::copy_to_build_dir(const std::string filename) -> bool {
```

Comment:
I'm not sure this is a good idea. The build-system supports using build directories other than `build`, so that assumption isn't stable.

Maybe it would make sense to keep this, but add an optional parameter to specify the build directory (defaulting that parameter to `build`)?

Reply:
Fixed this assumption wherever it is made according to directions given above.

DONE